libutils/string_lib.c

```diff
@@ -528,6 +528,27 @@ int StringToLong(const char *str, long *value_out)
     return 0;
 }
 
+/**
+ * @brief Convert a string representing a real number to int, discarding the decimal part.
+ *
+ * @param[in] str String matching the regex \d+(\.\d+)?, cannot be NULL
+ * @param[out] value_out Where to store result on success, cannot be NULL
+ * @return 0 on success, error code otherwise (see StringToLong source code)
+ */
+int StringDecimalToLong(const char *str, long *value_out)
+{
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    int len_of_int_part = strcspn(str, ".");
+    char int_part[len_of_int_part];
+
+    strncpy(int_part, str, len_of_int_part);
+    int_part[len_of_int_part] = '\0';
```

Comment:
Please make sure this doesn't do extra work if the string is just a simple integer.